deploy/scripts/src/authentication/test.ts

```diff
@@ -265,118 +273,180 @@ export function signUp (): void {
-        }))
+      group('B02_SignIn_05_AuthMFA_EnterTOTP POST', () => {
+        res = timeRequest(() => res.submitForm({
+          fields: { code: totp.generateTOTP() },
+          params: {
+            redirects: 1,
+            tags: { name: 'B02_SignIn_05_AuthMFA_EnterTOTP_01_AuthCall' }
+          }
+        }),
+        { isStatusCode302 })
+        res = timeRequest(() => http.get(res.headers.Location, {
+          redirects: 0,
+          tags: { name: 'B02_SignIn_05_AuthMFA_EnterTOTP_02_OIDCCall' }
+        }),
+        { isStatusCode302 })
+
+        rpStubCheck = (res.headers.Location).includes('auth-stub')
+        acceptNewTerms = (res.headers.Location).includes('signin')
+
+        if (rpStubCheck) {
+          res = timeRequest(() => http.get(res.headers.Location, {
+            tags: { name: 'B02_SignIn_05_AuthMFA_EnterTOTP_03_RPStub' }
+          }),
+          { isStatusCode200, ...pageContentCheck('User information') })
+        } else if (acceptNewTerms) {
+          res = timeRequest(() => http.get(res.headers.Location, {
+            tags: { name: 'B02_SignIn_05_AuthMFA_EnterTOTP_04_AuthAcceptTerms' }
+          }),
+          { isStatusCode200, ...pageContentCheck('terms of use update') })
+        }
+      })
       break
     }
     case 'SMS': {
-      res = group('B01_SignIn_07_SMSMFA_EnterPassword POST', () =>
+      res = group('B02_SignIn_06_SMSMFA_EnterPassword POST', () =>
         timeRequest(() =>
           res.submitForm({
             fields: { password: credentials.password },
-            params: { tags: { name: 'B01_SignIn_07_SMSMFA_EnterPassword' } }
+            params: { tags: { name: 'B02_SignIn_06_SMSMFA_EnterPassword' } }
           }), { isStatusCode200, ...pageContentCheck('Check your phone') }))
 
       sleep(1)
 
-      res = group('B01_SignIn_08_SMSMFA_EnterOTP POST', () =>
-        timeRequest(() => {
-          const response = res.submitForm({
-            fields: { code: credentials.phoneOTP },
-            params: { tags: { name: 'B01_SignIn_08_SMSMFA_EnterOTP' } }
-          })
-          acceptNewTerms = (response.body as string).includes('terms of use update')
-          return response
-        }, { isStatusCode200, 'verify page content': r => acceptNewTerms || (r.body as string).includes('User information') }))
+      group('B02_SignIn_07_SMSMFA_EnterOTP POST', () => {
+        res = timeRequest(() => res.submitForm({
+          fields: { code: credentials.phoneOTP },
+          params: {
+            redirects: 1,
+            tags: { name: 'B02_SignIn_07_SMSMFA_EnterOTP_01_AuthCall' }
+          }
+        }),
+        { isStatusCode302 })
+        res = timeRequest(() => http.get(res.headers.Location, {
+          redirects: 0,
+          tags: { name: 'B02_SignIn_07_SMSMFA_EnterOTP_02_OIDCCall' }
+        }),
+        { isStatusCode302 })
+
+        rpStubCheck = (res.headers.Location).includes('auth-stub')
+        acceptNewTerms = (res.headers.Location).includes('signin')
+
```

Comment:
Is this check sufficient to determine if the new terms and conditions screen is presented?

Reply:
Yes this check should be sufficient as this to checking the URL of the redirect. However, I have updated this to a slightly better check.